implementation/domain/logic/jbb-board-impl/src/main/java/org/jbb/board/impl/forum/logic/ForumServiceImpl.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (C) 2017 the original author or authors.
+ *
+ * This file is part of jBB Application Project.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  You may obtain a copy of the License at
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package org.jbb.board.impl.forum.logic;
+
+import org.apache.commons.lang3.Validate;
+import org.jbb.board.api.exception.ForumException;
+import org.jbb.board.api.model.Forum;
+import org.jbb.board.api.model.ForumCategory;
+import org.jbb.board.api.service.ForumService;
+import org.jbb.board.event.ForumCreatedEvent;
+import org.jbb.board.event.ForumRemovedEvent;
+import org.jbb.board.impl.forum.dao.ForumCategoryRepository;
+import org.jbb.board.impl.forum.dao.ForumRepository;
+import org.jbb.board.impl.forum.model.ForumCategoryEntity;
+import org.jbb.board.impl.forum.model.ForumEntity;
+import org.jbb.lib.eventbus.JbbEventBus;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.Validator;
+
+@Service
+public class ForumServiceImpl implements ForumService {
+    private final ForumRepository forumRepository;
+    private final ForumCategoryRepository categoryRepository;
+    private final JbbEventBus eventBus;
+    private final Validator validator;
+
+    @Autowired
+    public ForumServiceImpl(ForumRepository forumRepository, ForumCategoryRepository categoryRepository,
+                            JbbEventBus eventBus, Validator validator) {
+        this.forumRepository = forumRepository;
+        this.categoryRepository = categoryRepository;
+        this.eventBus = eventBus;
+        this.validator = validator;
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public Forum getForum(Long id) {
+        Validate.notNull(id);
+        return forumRepository.findOne(id);
+    }
+
+    @Override
+    @Transactional
+    public Forum addForum(Forum forum, ForumCategory category) {
```

Comment:
What will happen if I'll try add forum with name X which already exists in category Y?